lib/api/src/index.tsx

```diff
@@ -0,0 +1,211 @@
+import React, { Component } from 'react';
+
+import Events from '@storybook/core-events';
+import { Collection, Types } from '@storybook/addons';
+import initProviderApi from './init-provider-api';
+
+import Store from './store';
+import getInitialState from './initial-state';
+
+import initAddons from './modules/addons';
+import initChannel from './modules/channel';
+import initNotifications, { Notification } from './modules/notifications';
+import initStories, { StoriesHash } from './modules/stories';
+import initLayout from './modules/layout';
+import initShortcuts, { Shortcuts } from './modules/shortcuts';
+import initURL, { QueryParams } from './modules/url';
+import initVersions from './modules/versions';
+
+const ManagerContext = React.createContext({ api: undefined, state: getInitialState({}) });
+
+const { STORY_CHANGED, SET_STORIES, SELECT_STORY } = Events;
+
+interface RouterData {
+  location: Location;
+  path: string;
+  viewMode: 'story' | 'info' | string | undefined;
+  storyId: string;
+}
+
+export type Module = StoreData & RouterData & ProviderData & Navigate;
+
+interface Theme {
+  [key: string]: any;
+}
+
+export interface State extends RouterData {
+  layout: {
+    isFullscreen: boolean;
+    showPanel: boolean;
+    panelPosition: 'bottom' | 'right';
+    showNav: boolean;
+    isToolshown: boolean;
+  };
+
+  ui: {
+    name: string;
+    url: string;
+    enableShortcuts: boolean;
+    sortStoriesByKind: boolean;
+    sidebarAnimations: boolean;
+    theme: Theme;
+  };
+
+  storiesHash: StoriesHash;
+
+  shortcuts: Shortcuts;
+
+  customQueryParams: QueryParams;
+
+  notifications: Notification[];
+
+  versions: {
+    latest: any;
+    current: any;
+  };
+  lastVersionCheck: any;
+  dismissedVersionNotification: any;
+
+  [key: string]: any;
+}
+
+export interface API {
```

Comment:
Needs work later on, 

Can do that in a separate PR, as it's likely not super duper easy to get right, but obviously immensely useful!